### PR TITLE
Don't store documents non historical settings in order meta

### LINKED ISSUE #257: Don't store non-historical settings in order meta
This user reported an issue about the size of the meta stored by our plugin:
https://wordpress.org/support/topic/la-table-_wcpdf-est-tres-lourde/

I think 1kb per order could be about right for the meta we store for the invoice - it depends on how much text has been entered in the settings. While 1kb per order is not excessive, I do think we could be a bit more frugal, by filtering the array data before we store it, and removing all the [non-historical data](https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/blob/ec7b056ff3a4344f85ab641f477663bcade56848/includes/documents/abstract-wcpdf-order-document.php#L178-L188). Currently it even stores the `enabled`, `attach_to_email_ids` etc.

### PR DESCRIPTION
**Pull request by alexmigf**
closes #257

### CODE REVIEW

**Comment by Spreeuw** [comment]
The issue is merely about the _save_ process, and I think your approach is so different from the original that it opens a lot of other potential issues.

I propose that we focus on filtering the data being saved, and if you think there are other areas that need improvement, please open a separate issue for that, describing what you would like to improve.
A simple `array_filter()` would probably suffice here: https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/blob/3d765bd80cf2f6f35a2ab9eff661c389f7d8f4e9/includes/documents/abstract-wcpdf-order-document.php#L152-L154

**Review by Spreeuw — CHANGES_REQUESTED** [review]

File: includes/documents/abstract-wcpdf-order-document.php [hunk 1]
```diff
@@ -151,6 +151,9 @@ public function get_settings( $latest = false ) {
 		if ( $this->storing_settings_enabled() && empty( $order_settings ) && !empty( $this->order ) ) {
 			// this is either the first time the document is generated, or historical settings are disabled
 			// in both cases, we store the document settings
+			$non_historical_settings = $this->get_non_historical_settings();
+			// exclude non historical settings from order meta
+			$settings = array_diff_key( $settings, array_flip(  $non_historical_settings ) );
```
[the remark this answers is not in the document]
Comment (alexmigf) [reply includes/documents/abstract-wcpdf-order-document.php:156]: Ok

[the remark this answers is not in the document]
Comment (Spreeuw) [reply includes/documents/abstract-wcpdf-order-document.php]: @alexmigf You _still_ modify the `$settings` value, which is your _return_ value that should not be modified! See my comment above.

**Review by Spreeuw — APPROVED** [review]
Code looks good, I've added @YordanSoares to review for functional tests

**Review by YordanSoares — APPROVED** [review]
✔️ Looks good to me!

**Pull request merged**